### PR TITLE
Fix Windows runs on Github Actions

### PR DESCRIPTION
**Pull request by jackvreeken**
- [x] ~~It should not just be the `test_parse_cache_hit` test failing. It might be the _first_ to fail, but if we fix that one, I'd expect `test_parse_cache_purge` to fail.~~ The former is much quicker to do a second call compared to the latter (which reimports a module, so effective sleep)
- [x] Every hour or so, push two commits that undo and redo the sleep. Maybe alternating (00:00, 02:00,... ) we undo, and (01:00, ..) we redo. Do this for like a week or so to get some statistics.

Testing over in https://github.com/jackvreeken/pymoca-test-1 and https://github.com/jackvreeken/pymoca-test-2.

Current commit of this PR has been stress tested in https://github.com/jackvreeken/pymoca-test-2 for the last 12 hours (and will keep on running).

### CODE REVIEW

File: src/pymoca/parser.py
Comment (kentrutan) [inline src/pymoca/parser.py:1038]: This is a side issue, but I just noticed:

1. The `cutoff_time` calc should use `_microseconds_since_epoch()` instead of copy/paste code.
2. `datetime.utcnow()` used here and two other places is deprecated as of Python 3.12. They suggest using `datetime.now(datetime.timezone.utc)` instead.

**Review by kentrutan — APPROVED** [review]
Nice job tracking this one down! I read up a little on time precision and someone said it may actually be possible that a time could be less than a previous value on Windows, so there is a `time.monotonic_ns()` to prevent that. I don't know what `datetime` uses. Your fix should work either way though.

I'll let you merge this one when you are satisfied there has been enough test time on your fork.

File: src/pymoca/parser.py
Comment (jackvreeken) [reply src/pymoca/parser.py]: I added a bunch of commits on top to address both these

File: src/pymoca/parser.py
Comment (jackvreeken) [reply src/pymoca/parser.py:1044]: But +1 is exactly what we want right?

**Comment by jackvreeken** [comment]
> Nice job tracking this one down! I read up a little on time precision and someone said it may actually be possible that a time could be less than a previous value on Windows, so there is a `time.monotonic_ns()` to prevent that. I don't know what `datetime` uses. Your fix should work either way though.
> 
> I'll let you merge this one when you are satisfied there has been enough test time on your fork.

Hmm, we can use monotonic_ns as well, but it has a whopping 15 ms interval. I think we should just fall back on the `+1` logic then. I added a commit using `time.time_ns`  as a more accurate clock, to save some indirection (via datetime objects), and solve the deprecation of utcnow()

File: src/pymoca/parser.py
Comment (jackvreeken) [reply src/pymoca/parser.py:1044]: Understand now, I totally missed the + 1 at the end there. That's a remnant of me trying `? + 1` before realizing that `max()` works in SQLite.

**Pull request merged**
